src/matrixrtc/MatrixRTCSession.ts

```diff
@@ -892,6 +885,20 @@ export class MatrixRTCSession extends TypedEventEmitter<MatrixRTCSessionEvent, M
         }
     }
 
+    private stateEventsContainOngoingLegacySession(callMemberEvents: Map<string, MatrixEvent>) {
+        for (const callMemberEvent of callMemberEvents.values()) {
+            const content = callMemberEvent.getContent();
+            if (Array.isArray(content["memberships"])) {
+                for (const membership of content.memberships) {
+                    if (!(new CallMembership(callMemberEvent, membership).isExpired())) {
+                        return true;
```

Comment:
Changes from the suggestion:
- This iterates over `.values()` instead of `Array.from` to avoid the extra memory cost of creating a whole array out of an iterator
- This loops over `content.memberships` instead of using `.map` / `.filter` to allow early-returning as soon as a single ongoing session is found
- This negates `.isExpired()` because the goal is to find ongoing/non-expired sessions (right?)

Reply:
Cool. Makes sense.